sdk/src/metrics/export/periodic_exporting_metric_reader.cc

```diff
@@ -169,8 +169,8 @@ bool PeriodicExportingMetricReader::OnForceFlush(std::chrono::microseconds timeo
   is_force_flush_notified_.store(false, std::memory_order_release);
   if (result)
   {
-
-    result = exporter_->ForceFlush(timeout);
+    result = exporter_->ForceFlush(
+        std::chrono::duration_cast<std::chrono::microseconds>(timeout_steady));
```

Comment:
I think we should check and ensure the `timeout_steady` to be greater than zero, some exporters use zero and negative value as indefinitely waiting.

Reply:
What's your suggestion - 
 - if the original `timeout` as argument to period-reader::forceflush is zero, we can pass that directly to the exporter::forceflush, irrespective of the value of `timeout_steady`.
 - else if the timeout_steady is zero or less, we can return error, and don't call exporter::forceflush
 - else call exporter::forceflush with non-zero timeout_steady

Reply:
Looks good.

Reply:
Thanks, have updated the code as per above logic, let me know if that looks fine :)